### PR TITLE
GEM-7 | Add integration tests and refactor

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,6 +44,13 @@ jobs:
...
 
       - name: Run tests
-        run: ./gradlew test
+        run: ./gradlew clean check
+
+      - name: Publish Test Report
+        uses: mikepenz/action-junit-report@v3
+        if: success() || failure()
+        with:
+          report_paths: '**/build/test-results/test/TEST-*.xml'
```
Comment (pwgacek) [inline .github/workflows/ci.yml:53]: is this error intended behavior?
Screenshot from 2024-03-26 19-15-23

File: src/test/kotlin/pl/edu/agh/gem/util/TestHelper.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    name = name,
+)
+
+fun createProductRequest(
+    name: String = "name",
+) = ProductRequest(
+    name = name,
+)
+
+fun createExampleProductRespone(
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/util/TestHelper.kt]: typo

File: src/test/kotlin/pl/edu/agh/gem/internal/service/ProductServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package pl.edu.agh.gem.internal.service
+
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.should
```
Comment (pwgacek) [inline src/test/kotlin/pl/edu/agh/gem/internal/service/ProductServiceTest.kt]: unused import. weird ktlint did not find it

File: settings.gradle.kts [hunk 4]
```diff
@@ -1 +1,85 @@
-rootProject.name = "application"
+rootProject.name = "exampleService"
```
Comment (pwgacek) [inline settings.gradle.kts]: think we should use the name of repo

File: src/main/kotlin/pl/edu/agh/gem/external/persistence/MongoProductRepository.kt [hunk 5]
```diff
@@ -1,13 +1,33 @@
 package pl.edu.agh.gem.external.persistence
 
 import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.findOne
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/persistence/MongoProductRepository.kt]: unused import

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/integration/kotlin/pl/edu/agh/gem/integration/controler/ProductControllerIT.kt [hunk 6]
```diff
@@ -40,4 +45,26 @@ class ProductControllerIT(
...
+    should("throw error when product with given id does not exist") {
+        // given
+        val productRequest = createProductRequest(name = "")
+
+        // when
+        val response = service.createProduct(productRequest)
+
+        // then
+        response shouldHaveHttpStatus BAD_REQUEST
+        println("=====================================")
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/controler/ProductControllerIT.kt]: print

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/integration/kotlin/pl/edu/agh/gem/integration/controler/ProductControllerIT.kt [hunk 7]
```diff
@@ -0,0 +1,67 @@
+package pl.edu.agh.gem.integration.controler
+
+import io.kotest.matchers.nulls.shouldBeNull
+import io.kotest.matchers.shouldBe
+import org.springframework.http.HttpStatus.BAD_REQUEST
+import org.springframework.http.HttpStatus.OK
+import pl.edu.agh.gem.external.controller.SimpleErrorsHolder
```
Comment (lawos98) [inline src/integration/kotlin/pl/edu/agh/gem/integration/controler/ProductControllerIT.kt]:
```suggestion
```

**Review by pwgacek — APPROVED** [review]

**Pull request merged**